packages/editor-ui/src/components/ExpressionParameterInput.vue

```diff
@@ -0,0 +1,297 @@
+				{{ $locale.baseText('parameterInput.resultForItem') }} {{ hoveringItemNumber }}
+			</n8n-text>
+			<n8n-text :class="$style.body">
+				<InlineExpressionEditorOutput
+					:value="value"
+					:isReadOnly="isReadOnly"
+					:segments="segments"
+				/>
+			</n8n-text>
+			<div :class="$style.footer">
+				<n8n-text size="small" compact>
+					{{ $locale.baseText('parameterInput.anythingInside') }}
+				</n8n-text>
+				<div :class="$style['expression-syntax-example']" v-text="`{{ }}`"></div>
+				<n8n-text size="small" compact>
+					{{ $locale.baseText('parameterInput.isJavaScript') }}
+				</n8n-text>
+				<n8n-link
+					:class="$style['learn-more']"
+					size="small"
+					underline
+					theme="text"
+					:to="expressionsDocsUrl"
+				>
+					{{ $locale.baseText('parameterInput.learnMore') }}
+				</n8n-link>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { mapStores } from 'pinia';
+import Vue from 'vue';
+import mixins from 'vue-typed-mixins';
+
+import { useNDVStore } from '@/stores/ndv';
+import InlineExpressionEditorInput from '@/components/InlineExpressionEditor/InlineExpressionEditorInput.vue';
+import InlineExpressionEditorOutput from '@/components/InlineExpressionEditor/InlineExpressionEditorOutput.vue';
+import ExpressionFunctionIcon from '@/components/ExpressionFunctionIcon.vue';
+import { telemetryUtils } from '@/mixins/telemetryUtils';
+import { EXPRESSIONS_DOCS_URL, LOCAL_STORAGE_MAIN_PANEL_RELATIVE_WIDTH } from '@/constants';
+
+import type { Segment } from '@/types/expressions';
+import type { TargetItem } from '@/Interface';
+
+Vue.component('ExpressionFunctionIcon', ExpressionFunctionIcon);
+
+export default mixins(telemetryUtils).extend({
+	name: 'ExpressionParameterInput',
+	components: {
+		InlineExpressionEditorInput,
+		InlineExpressionEditorOutput,
+	},
+	data() {
+		return {
+			isFocused: false,
+			segments: [] as Segment[],
+			paneWidth: -1,
+			expressionsDocsUrl: EXPRESSIONS_DOCS_URL,
+		};
+	},
+	props: {
+		value: {
+			type: String,
+		},
+		isReadOnly: {
+			type: Boolean,
+			default: false,
+		},
+		isForRecordLocator: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	computed: {
+		...mapStores(useNDVStore),
+		hoveringItemNumber(): number {
+			return (this.hoveringItem?.itemIndex ?? 0) + 1;
+		},
+		hoveringItem(): TargetItem | null {
+			return this.ndvStore.hoveringItem;
+		},
+		isDragging(): boolean {
+			return this.ndvStore.isDraggableDragging;
+		},
+		currentNodePaneType(): string {
+			return this.ndvStore.currentNodePaneType;
+		},
+	},
+	methods: {
+		focus() {
+			const inlineInput = this.$refs.inlineInput as (Vue & HTMLElement) | undefined;
+
+			if (inlineInput?.$el) inlineInput.focus();
+		},
+		onFocus() {
+			this.isFocused = true;
+
+			this.paneWidth = this.getPaneWidth();
+
+			this.$emit('focus');
+		},
+		getPaneWidth() {
+			const key = [LOCAL_STORAGE_MAIN_PANEL_RELATIVE_WIDTH, this.currentNodePaneType].join('_');
```

Comment:
you can see another approach of how to set this here, which might a little better https://github.com/n8n-io/n8n/blob/master/packages/editor-ui/src/components/ResourceLocator/ResourceLocator.vue#L433

Reply:
The only purpose of tracking pane width was to prevent blur on resize, but seeing as this requires adding to state, watching, comparing to prior width, and possible issues with NDV-to-NDV jumps, I [refactored this](https://github.com/n8n-io/n8n/pull/4814/commits/dda84915e80843bdc90c0b6b2ef5dd8c6d48aa26) to only rely only on the presence of the `resizer` class. This no longer tracks anything and does not fire events on resize, which also (apparently?) even makes dragging to resize snappier than before because we are no longer triggering state changes. Let me know what you think.